include/dsn/service_api_c.h

```diff
@@ -500,7 +500,7 @@ extern DSN_API bool         dsn_semaphore_wait_timeout(
 //------------------------------------------------------------------------------
 
 // rpc address utilities
-
+extern DSN_API int           dsn_address_use_ip_as_name;
```

Comment:
One way is to simply remove the other case - always print ip addresses in logs. I'm considering removing IPV6 in dsn_address_t next, and make sure the c address can fit into a 64bit integer, which simplifies many things such as passing as value across functions instead of pointers etc. And you primary_address related changed can also be simplified a lot.